src/ray/raylet/node_manager.cc

```diff
@@ -539,6 +545,66 @@ ray::Status NodeManager::RegisterGcs() {
         "NodeManager.deadline_timer.print_event_loop_stats");
   }
 
+  if (RayConfig::instance().use_ray_syncer()) {
+    // Register resource manager and scheduler
+    ray_syncer_.Register(
+        syncer::RayComponentId::RESOURCE_MANAGER,
+        this,
```

Comment:
Can you add the arg name like `/*reporter*/` for each line?

Reply:
IIRC the GCS sends cluster information of all nodes to all raylets before, but the raylet will ignore the information belongs to itself.
If we keep it this way, does it make the protocol simpler?

Reply:
@WangTaoTheTonic it'll only work for the current setup. The idea here is to decouple syncer with gcs/raylet so that we can do an extension in the future.

What you said won't work given a tree/dag/graph structure which we'll explore in the future to reduce loads of gcs.